samples/generate-offline-map-with-custom-parameters/src/main/java/com/esri/arcgismaps/sample/generateofflinemapwithcustomparameters/components/GenerateOfflineMapWithCustomParametersViewModel.kt

```diff
@@ -0,0 +1,386 @@
+    val snackbarHostState = SnackbarHostState()
+
+    // Determinate job progress loading dialog visibility state
+    var showJobProgressDialog by mutableStateOf(false)
+        private set
+
+    // Determinate job progress percentage
+    var offlineMapJobProgress by mutableIntStateOf(0)
+        private set
+
+    var showResetButton by mutableStateOf(false)
+        private set
+
+    private var generateOfflineMapJob: GenerateOfflineMapJob? = null
+
+    // Create an IntSize to retrieve dimensions of the map
+    private var mapViewSize by mutableStateOf(IntSize(0, 0))
+
+    fun updateMapViewSize(size: IntSize) {
+        mapViewSize = size
+    }
+
+    init {
+        setUpMap()
+    }
+
+    /**
+     * Use map view's size to determine dimensions of the map to get the download offline area
+     * and use [MapViewProxy] to assist in converting screen points to map points
+     */
+    fun calculateDownloadOfflineArea() {
+        // Ensure the map is loaded before calculating the download area
+        if (arcGISMap.loadStatus.value == LoadStatus.Loaded) {
+            // Upper left corner of the area to take offline
+            val minScreenPoint = ScreenCoordinate(200.0, 200.0)
+            // Lower right corner of the downloaded area
+            val maxScreenPoint = ScreenCoordinate(
+                x = mapViewSize.width - 200.0, y = mapViewSize.height - 200.0
+            )
+            // Convert screen points to map points
+            val minPoint = mapViewProxy.screenToLocationOrNull(minScreenPoint)
+            val maxPoint = mapViewProxy.screenToLocationOrNull(maxScreenPoint)
+            // Create an envelope to set the download area's geometry using the defined bounds
+            if (minPoint != null && maxPoint != null) {
+                val envelope = Envelope(minPoint, maxPoint)
+                downloadAreaGraphic.geometry = envelope
+            }
+        }
+    }
+
+    /**
+     * Sets up a portal item and displays map area to take offline
+     */
+    private fun setUpMap() {
+        // Create a portal item with the itemId of the web map
+        val portal = Portal("https://www.arcgis.com")
+        val portalItem = PortalItem(portal, "acc027394bc84c2fb04d1ed317aac674")
+
+        // Clear, then add the download graphic to the graphics overlay
+        graphicsOverlay.graphics.clear()
+        graphicsOverlay.graphics.add(downloadAreaGraphic)
+
+        arcGISMap = ArcGISMap(portalItem)
+        viewModelScope.launch(Dispatchers.Main) {
+            arcGISMap.load().onFailure {
+                messageDialogVM.showMessageDialog(
+                    title = it.message.toString()
+                )
+            }
+        }
+        showResetButton = false
+    }
+
+    /**
+     * Define the [GenerateOfflineMapParameters] for the offline map job and add the custom
+     * [GenerateOfflineMapParameterOverrides] using the given override values.
+     */
+    fun defineGenerateOfflineMapParameters(
```

Comment:
I would remove the function chaining in this function and instead capture the results on a variable. 
It would improve readability and allow you to perform an action when the `create` method fails. 
For example: 
```
  // Create default generate offline map parameters from the offline map task
  val generateOfflineMapParameters =
      offlineMapTask.createDefaultGenerateOfflineMapParameters(areaOfInterest = downloadArea).getOrElse {
          messageDialogVM.showMessageDialog(
              title = "Error", description = "Failed to create default generate offline map parameters"
          )
          return@launch
      }
  // Return a job failure if generate offline map encounters an error
  generateOfflineMapParameters.apply {
      continueOnErrors = false
  }
```